viewer/packages/360-images/src/Image360Facade.ts

```diff
@@ -56,12 +70,65 @@ export class Image360Facade<T> {
     camera: THREE.Camera
   ): Image360Entity | undefined {
     this._rayCaster.setFromCamera(coords, camera);
-    const intersections = this._rayCaster.intersectObjects<Image360Icon>(
-      this._image360Entities.map(p => p.icon),
-      false
-    );
-    if (intersections.length > 0) {
-      return this._image360Entities.filter(p => p.icon === intersections[0].object)[0];
+    const cameraDirection = camera.getWorldDirection(new THREE.Vector3());
+    const cameraPosition = camera.position.clone();
+
+    const intersections = this._image360Collections
+      .flatMap(getImage360Entities)
+      .filter(hasVisibleIcon)
+      .map(entity => getIntersection(entity, this._rayCaster.ray))
+      .filter(hasIntersection)
+      .map(intersectionToCameraSpace)
+      .filter(isInFrontOfCamera)
+      .sort(byDistanceToCamera)
+      .map(selectEntity);
```

Comment:
Yeees, I like this so much :drooling_face:

Reply:
<img src="https://media0.giphy.com/media/PjsPhYsS2WDO5e0RaO/giphy.gif"/>